components/calendar/src/chinese.rs

```diff
@@ -106,18 +110,20 @@ pub struct Chinese;
 
 /// The inner date type used for representing [`Date`]s of [`Chinese`]. See [`Date`] and [`Chinese`] for more details.
 #[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
-pub struct ChineseDateInner(ArithmeticDate<Chinese>);
+pub struct ChineseDateInner(ChineseBasedDateInner<Chinese>);
+
+type Inner = ChineseBasedDateInner<Chinese>;
 
 impl CalendarArithmetic for Chinese {
     /// Returns the number of days in the given (year, month). In the Chinese calendar, months start at each
     /// new moon, so this function finds the number of days between the new moon at the beginning of the given
     /// month and the new moon at the beginning of the next month.
     fn month_days(year: i32, month: u8) -> u8 {
```

Comment:
suggestion: you should also move this impl over into ChineseBased; by using `impl<C: ChineseBased> CalendarArithmetic for C`

Reply:
Done